File: WebSiteRoute/Controllers/HomeController.cs
using Microsoft.AspNetCore.Mvc;
using System.Diagnostics;
using WebSiteRoute.Application.FlightAltitudes;
using WebSiteRoute.Models;
using WebSiteRoute.Presentation.ViewModels;

namespace WebSiteRoute.Controllers;

public class HomeController(ILogger<HomeController> logger, GetFlightAltitudeDashboardQuery dashboardQuery) : Controller
{
    public async Task<IActionResult> Index(CancellationToken cancellationToken)
    {
        var dashboard = await dashboardQuery.ExecuteAsync(cancellationToken);
        var viewModel = FlightAltitudeDashboardViewModel.FromDashboard(dashboard);

        logger.LogInformation("Rendered flight status message for altitude {Altitude}", dashboard.SampleAltitude);

        return View(viewModel);
    }

    public IActionResult Privacy()
    {
        return View();
    }

    [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
    public IActionResult Error()
    {
        return View(new ErrorViewModel { RequestId = Activity.Current?.Id ?? HttpContext.TraceIdentifier });
    }
}

File: src/main/java/com/websiteroute/presentation/HomeController.java
package com.websiteroute.presentation;

import com.websiteroute.application.GetFlightAltitudeDashboardQuery;
import org.springframework.stereotype.Controller; import org.springframework.ui.Model; import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class HomeController {
    private final GetFlightAltitudeDashboardQuery query;
    public HomeController(GetFlightAltitudeDashboardQuery query) { this.query = query; }
    @GetMapping({"/", "/home"}) public String index(Model model) { model.addAttribute("dashboard", query.execute()); return "index"; }
    @GetMapping("/privacy") public String privacy() { return "privacy"; }
}
